public/js/index.js

```diff
@@ -524,6 +524,62 @@ Index.loadContextMenuCategories = (catList, id) => Server.callAPI(`/api/archives
     },
 );
 
+/**
+ * Build rating options for contextMenu and select the one for the current ID.
+ * @param {*} id The ID of the archive to check
+ * @returns Ratings
+ */
+Index.loadContextMenuRatings = (id) => Server.callAPI(`/api/archives/${id}/metadata`, "GET", null, `Error finding metadata for ${id}!`,
+    (data) => {
+        const items = {};
+        const ratings = [{
+            id: 1,
+            name: "⭐",
+        }, {
+            id: 2,
+            name: "⭐⭐",
+        }, {
+            id: 3,
+            name: "⭐⭐⭐",
+        }, {
+            id: 4,
+            name: "⭐⭐⭐⭐",
+        }, {
+            id: 5,
+            name: "⭐⭐⭐⭐⭐",
+        }];
+        const tags = data.tags.split(",").map((x) => x.split(":"));
+        const hasRating = tags.some((x) => x[0] === "rating");
+
+        items.push({
+            name: "",
+            events: {
+                click() {
+                    Server.updateTagsFromArchive(id, tags.filter((x) => x[0] !== "rating").map((x) => x.join(":")).join(","));
+                },
+            },
+        });
+
+        for (let i = 0; i < ratings.length; i++) {
+            items[i] = ratings[i];
+            items[i].events = {
+                click() {
+                    if (hasRating) {
```

Comment:
It would be nice to show in the UI the current rating, if there already was one.  
Maybe a checkmark can be added to the matching item or something?

Reply:
Fixed on new commit.